src/log.rs

```diff
@@ -178,13 +178,11 @@ pub type itxfm_fn = Option<
     unsafe extern "C" fn(*mut libc::c_void, ptrdiff_t, *mut libc::c_void, libc::c_int) -> (),
 >;
 pub type coef = ();
+use crate::src::mc::Dav1dMCDSPContextRust;
 #[derive(Copy, Clone)]
 #[repr(C)]
 pub struct Dav1dMCDSPContext {
-    pub mc: [mc_fn; 10],
-    pub mc_scaled: [mc_scaled_fn; 10],
-    pub mct: [mct_fn; 10],
-    pub mct_scaled: [mct_scaled_fn; 10],
+    pub rust: Dav1dMCDSPContextRust,
```

Comment:
Why do we need to have types and fields named `rust` and `Dav1dMCDSPContextRust`? We should name stuff based on its purpose (what it does) not the implementation language (how it does it).

Reply:
They're just temporary.  I wanted to only devirtualize a few fields of that struct to start with, but once the rest are, too, these Rust-named structs and fields will go away.  Is there another name you think would be good?

Reply:
I think it would be good to deduplicate `Dav1dMCDSPContext` first (can land as a separate PR) and then just do to the outer/original structure what you want. This PR isn't easy to review to begin with and the introduction of temporary structures doesn't make it easier. 

If you must  – I'd advise advise against temporary changes if only done for convenience since it makes for a harder to understand commit history – please *always* label temporary stuff like so `// TODO(kkysen): Remove once ...`. This has two benefits:

1. It leaves our future selves with breadcrumbs so we don't miss stuff we need to cleanup
2. It communicates to others what your plans are. I won't nitpick or worry so much about stuff that goes away down the line.

Reply:
I'll definitely add a comment to make it clear it's temporary.  We can't deduplicate it first, though.  The devirtualization is needed to deduplicate it since it gets rid of the bitdepth-dependent function pointers, which make deduplication extremely difficult.

The other alternative is to avoid the temporary is to devirtualize the whole struct at once, but that would be a very large PR so I wanted to split it up.